transferui/src/main/java/com/hyperwallet/android/ui/transfer/view/ListTransferDestinationActivity.java

```diff
@@ -0,0 +1,109 @@
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+package com.hyperwallet.android.ui.transfer.view;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.View;
+import android.view.WindowManager;
+
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.appcompat.widget.Toolbar;
+import androidx.core.content.ContextCompat;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentManager;
+import androidx.fragment.app.FragmentTransaction;
+import androidx.lifecycle.Observer;
+import androidx.lifecycle.ViewModelProviders;
+
+import com.hyperwallet.android.model.transfermethod.HyperwalletTransferMethod;
+import com.hyperwallet.android.ui.common.repository.Event;
+import com.hyperwallet.android.ui.transfer.R;
+import com.hyperwallet.android.ui.transfer.viewmodel.ListTransferDestinationViewModel;
+
+public class ListTransferDestinationActivity extends AppCompatActivity {
+
+    public static final String ARGUMENT_SELECTED_DESTINATION = "argument_selected_destination";
+    private static final String TAG = ListTransferDestinationActivity.class.getSimpleName();
+
+    private ListTransferDestinationViewModel mListTransferDestinationViewModel;
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_list_transfer_destination);
+
+        Toolbar toolbar = findViewById(R.id.toolbar);
+        toolbar.setTitle(R.string.destination);
+        setSupportActionBar(toolbar);
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+        getSupportActionBar().setDisplayShowHomeEnabled(true);
+        toolbar.setNavigationOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                finish();
+            }
+        });
+
+        if (savedInstanceState == null) {
+            initFragment(ListTransferDestinationFragment.newInstance(
+                    (HyperwalletTransferMethod) getIntent().getParcelableExtra(ARGUMENT_SELECTED_DESTINATION)));
+        }
+
+        mListTransferDestinationViewModel = ViewModelProviders.of(this).get(
```

Comment:
ModelView can't be initialized like this, it will have null pointer exception while calling platform for Transfer method list
please use the provided Factory `ListTransferDestinationViewModel.ListTransferDestinationViewModelFactory` to initialize viewmodel

Reply:
fixed